lib/internal/repl.js

```diff
@@ -51,19 +53,40 @@ function createRepl(env, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY_FILE) {
-    return setupHistory(repl, env.NODE_REPL_HISTORY_FILE, cb);
+  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
+    return setupHistory(repl, env.NODE_REPL_HISTORY,
+                        env.NODE_REPL_HISTORY_FILE, cb);
   }
   repl._historyPrev = _replHistoryMessage;
   cb(null, repl);
 }
 
-function setupHistory(repl, historyPath, ready) {
-  const fs = require('fs');
+function setupHistory(repl, historyPath, oldHistoryPath, ready) {
+  if (!historyPath) {
+    historyPath = path.join(os.homedir(), '.node_repl_history');
```

Comment:
Oh, whoops, I had forgotten about that, thanks.

It'd probably depend on the error what we may want to do; there a difference between say `ENOMEM` and `ENOENT`. :P